hypothesis-python/src/hypothesis/extra/pytestplugin.py

```diff
@@ -179,8 +180,19 @@ def pytest_collection_modifyitems(items):
     for item in items:
         if not isinstance(item, pytest.Function):
             continue
-        if getattr(item.function, 'is_hypothesis_test', False):
+        if is_hypothesis_test(item.obj):
             item.add_marker('hypothesis')
+        if getattr(item.obj, 'is_hypothesis_strategy_function', False):
+            def note_strategy_is_not_test(*args, **kwargs):
+                note_deprecation(
+                    '%s is a Hypothesis strategy, not a test.  This is '
```

Comment:
This error message feels slightly odd to me. I'm not sure why. Maybe something like the following would be better?

> %s defines a Hypothesis strategy, not a test. Perhaps you meant to use `@given` instead of `@composite`?

Reply:
Ah, in hindsight it was written for a particular user.  It should really describe what has happened instead, e.g. 

> %s is a function that returns a Hypothesis strategy, but Pytest has collected it as a test function (probably because of the name).  To define a test function, use `@given`.  You can draw examples from strategies inside the test with the `st.data()` strategy.